packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -203,19 +221,31 @@ contract Attestations is
 
     if (accounts[msg.sender].attestationRequestFeeToken != address(0x0)) {
       require(
-        !isAttestationTimeValid(accounts[msg.sender].mostRecentAttestationRequest) ||
+        !isAttestationRequestBlockTimeValid(accounts[msg.sender].mostRecentAttestationRequest) ||
           accounts[msg.sender].attestationRequestFeeToken == attestationRequestFeeToken,
         "A different fee token was previously specified for this account"
       );
     }
 
+    IdentifierState storage state = identifiers[identifier];
+    if(state.requests[msg.sender].blockNumber > 0) {
+      // TODO: This should be bound by Random#randomnessBlockRetentionWindow once that gets merged
+      require(
```

Comment:
Is this check redundant with the one on line 224?

Reply:
These checks are slightly different even though they looks quite similar. 224 checks that the `attestationFeeToken` cannot be changed while attestations can be completed. This check ensures that one does not request Attestations while not having selected issuers for a previous request

Reply:
Can they be combined? Something like:

```
if(!isAttestationExpired(accounts[msg.sender].mostRecentAttestationRequest)) {
  require(state.requests[msg.sender].blockNumber == 0, "Cannot request more attestations without first selecting issuers for a previous, unexpired attestation request");
  require(accounts[msg.sender].attestationRequestFeeToken == address(0) || accounts[msg.sender].attestationRequestFeeToken == attestationRequestFeeToken, "Cannot request more attestations with a different fee token without waiting for the most recent attestation request to expire");
}
```